internal/bminventory/inventory.go

```diff
@@ -4371,6 +4375,19 @@ func (b *bareMetalInventory) GetClusterByKubeKey(key types.NamespacedName) (*com
 }
 
 func (b *bareMetalInventory) GetClusterHostRequirements(ctx context.Context, params installer.GetClusterHostRequirementsParams) middleware.Responder {
-	// TODO: implement
-	return installer.NewGetClusterHostRequirementsOK().WithPayload(models.ClusterHostRequirementsList{})
+
+	cluster, err := b.getCluster(ctx, params.ClusterID.String(), common.UseEagerLoading)
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+	requirementsList := models.ClusterHostRequirementsList{}
+	for _, clusterHost := range cluster.Hosts {
```

Comment:
Instead of returning the requirements per host, why not return the information per role? Taking into account that there are a limited set of roles available, iterating over the hosts means that the information in this payload will probably be replicated several times over.

Reply:
The model design was a result of couple of discussions but maybe that would be an acceptable approach as well.
@avishayt - WDYT? Would it be OK to change the model to return requirements per role and not per host? That will require UI to make additional matching.